pandas/core/internals/blocks.py

```diff
@@ -1547,14 +1530,13 @@ def quantile(self, qs, interpolation="linear", axis=0):
             # We need to operate on i8 values for datetimetz
             # but `Block.get_values()` returns an ndarray of objects
             # right now. We need an API for "values to do numeric-like ops on"
-            values = self.values.asi8
+            values = self.values.view("M8[ns]")
```

Comment:
we should probably remove .asi8 on Index as I agree its sligthly confusing (I don't think we need deprecation)